clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1595,4 +1595,105 @@ else if (recordPartition != null)
             return topicPartition;
         }
     }
+
+    /**
+     * A class that represents the state of a prepared transaction that can be
+     * serialized to and deserialized from a string.
+     * Objects of this class can be written to and read from a database
+     * to support recovery of prepared transactions.
+     */
+    public static class PreparedTxnState {
+        private final long producerId;
+        private final short epoch;
+        
+        /**
+         * Creates a new empty PreparedTxnState
+         */
+        public PreparedTxnState() {
+            this.producerId = -1L;
+            this.epoch = -1;
```

Comment:
We can use `RecordBatch.NO_PRODUCER_ID` and `RecordBatch.NO_PRODUCER_EPOCH` here.